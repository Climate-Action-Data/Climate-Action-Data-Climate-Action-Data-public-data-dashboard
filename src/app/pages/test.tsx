'use client'
import { type NextPage } from "next";

const Test: NextPage = () => {
    return (
        <div>
            <h1>Hello</h1>
            <p>data</p>
        </div>
    )
}

export default Test