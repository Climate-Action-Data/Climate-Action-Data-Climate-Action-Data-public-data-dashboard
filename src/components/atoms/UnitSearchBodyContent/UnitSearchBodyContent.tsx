import { changeHoverColor } from '@/utils/Stickify'
import { Table, Tbody, Td, Tr } from '@chakra-ui/react'
import { EffectResponse } from '@/@types/EffectResponse'
import { useRouter } from 'next/navigation'
import { useTranslation } from 'react-i18next'
import { ProjectSearchBodySkeleton } from '../ProjectSearchBodySkeleton/ProjectSearchBodySkeleton'
import { UnitSearchResponse, UnitSearchResult } from '@/@types/UnitSearchResult'
import { formatDate } from '@/utils/DateFormat'
import { DateFormats } from '@/@types/DateFormats'
import { TableData } from '../TableData/TableData'
import { generateUnitUrl } from '@/utils/RequestHelpers'

interface UnitSearchBodyContentProps {
  unitResults?: EffectResponse<UnitSearchResponse>
}

export const UnitSearchBodyContent = (props: UnitSearchBodyContentProps) => {
  const { unitResults } = props
  const router = useRouter()
  const { t } = useTranslation(`home`)

  const generateTableRow = (unitList: UnitSearchResult[]) => {
    return unitList.map((unit, idx) => {
      const generatedUrl = generateUnitUrl(`${unit?.status}`)
      return (
        <Tr
          onClick={() => router.push(`${generatedUrl}${unit.warehouseUnitId}`)}
          data-testid="unit-table-row"
          onMouseEnter={() => changeHoverColor(`project-row-${idx}`, `hoverGreen`)}
          className={`project-row-${idx}`}
          key={`project-body-row-${unit.warehouseUnitId}`}
          height="92px"
        >
          <Td title={unit.status}>
            <TableData data={unit.status} />
          </Td>
          <Td title={unit.vintage?.toString()}>
            <TableData data={unit.vintage?.toString()} />
          </Td>
          <Td isNumeric title={unit.annualEst?.toLocaleString()}>
            <TableData data={unit.annualEst?.toLocaleString()} />
          </Td>
          <Td title={unit?.issuanceDate ? formatDate(unit.issuanceDate, DateFormats.YYYY_MM_DD) : t(`noData`)}>
            <TableData data={unit?.issuanceDate ? formatDate(unit.issuanceDate, DateFormats.YYYY_MM_DD) : t(`noData`)} />
          </Td>
          <Td title={unit?.retirementDate ? formatDate(unit.retirementDate, DateFormats.YYYY_MM_DD) : t(`noData`)}>
            <TableData data={unit?.retirementDate ? formatDate(unit.retirementDate, DateFormats.YYYY_MM_DD) : t(`noData`)} />
          </Td>
          <Td title={unit?.project?.standard}>
            <TableData data={unit?.project?.standard} />
          </Td>
          <Td title={unit?.project?.methodology}>
            <TableData data={unit?.project?.methodology} />
          </Td>
          <Td title={unit?.project?.sector}>
            <TableData data={unit?.project?.sector} />
          </Td>
          <Td title={unit?.project?.country}>
            <TableData data={unit?.project?.country} />
          </Td>
          <Td title={unit?.correspondingAdjustment}>
            <TableData data={unit?.correspondingAdjustment} />
          </Td>
          <Td minW="250px !important" title={unit.marketplace}>
            <TableData data={unit.marketplace} />
          </Td>
          <Td minW="250px !important" title={unit.serialNumber}>
            <TableData data={unit.serialNumber} />
          </Td>
        </Tr>
      )
    })
  }

  return (
    <Table variant="simple" className="searchTable">
      <Tbody data-testid="table-scroll">{unitResults?.data?.units !== undefined ? generateTableRow(unitResults.data.units) : <ProjectSearchBodySkeleton />}</Tbody>
    </Table>
  )
}
