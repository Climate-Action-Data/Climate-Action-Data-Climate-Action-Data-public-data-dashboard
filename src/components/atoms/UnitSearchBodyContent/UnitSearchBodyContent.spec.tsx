import { fireEvent, render, screen } from '@testing-library/react'

import { UnitSearchBodyContent } from './UnitSearchBodyContent'
import { UNIT_SEARCH_RESULT } from '@/test/TestOvermindMockData'
import { TestRouter } from '@/test/TestRouter'

it(`renders correctly`, () => {
  const { container } = render(
    <TestRouter router={{}}>
      <UnitSearchBodyContent unitResults={UNIT_SEARCH_RESULT.unitResults} />
    </TestRouter>,
  )
  expect(container).toMatchSnapshot()
})

it(`renders correctly with hover`, () => {
  const { container } = render(
    <TestRouter router={{}}>
      <UnitSearchBodyContent unitResults={UNIT_SEARCH_RESULT.unitResults} />
    </TestRouter>,
  )
  const rows = screen.getAllByTestId(`unit-table-row`)
  expect(rows.length).toBeGreaterThan(0)
  fireEvent.mouseOver(rows[0])
  expect(container).toMatchSnapshot()
})
