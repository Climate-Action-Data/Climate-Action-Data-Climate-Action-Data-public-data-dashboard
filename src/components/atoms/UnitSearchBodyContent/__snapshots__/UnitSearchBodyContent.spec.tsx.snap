// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <table
    class="chakra-table searchTable css-0"
  >
    <tbody
      class="css-0"
      data-testid="table-scroll"
    >
      <tr
        class="project-row-0 css-mii0oe"
        data-testid="unit-table-row"
      >
        <td
          class="css-0"
          title="Held"
        >
          Held
        </td>
        <td
          class="css-0"
          title="2020"
        >
          2020
        </td>
        <td
          class="css-0"
          title="129,037"
        >
          129,037
        </td>
        <td
          class="css-0"
          title="2023/04/14"
        >
          2023/04/14
        </td>
        <td
          class="css-0"
          title="EcoRegistry"
        >
          EcoRegistry
        </td>
        <td
          class="css-0"
          title="AR-ACM0003: Afforestation and reforestation of lands except wetlands"
        >
          AR-ACM0003: Afforestation and reforestation of lands except wetlands
        </td>
        <td
          class="css-0"
          title="Agriculture; forestry and fishing"
        >
          Agriculture; forestry and fishing
        </td>
        <td
          class="css-0"
          title="Colombia"
        >
          Colombia
        </td>
        <td
          class="css-0"
          title="Not Started"
        >
          Not Started
        </td>
        <td
          class="css-rl3ocm"
        >
          <p
            class="chakra-text css-132cox5"
          >
            --
          </p>
        </td>
        <td
          class="css-rl3ocm"
          title="CDC_29_7_1_321_14_CO_1_3_2020_BF_1-CDC_29_7_1_321_14_CO_1_3_2020_BF_4936"
        >
          CDC_29_7_1_321_14_CO_1_3_2020_BF_1-CDC_29_7_1_321_14_CO_1_3_2020_BF_4936
        </td>
      </tr>
      <tr
        class="project-row-1 css-mii0oe"
        data-testid="unit-table-row"
      >
        <td
          class="css-0"
          title="Retired"
        >
          Retired
        </td>
        <td
          class="css-0"
          title="2019"
        >
          2019
        </td>
        <td
          class="css-0"
          title="20,394"
        >
          20,394
        </td>
        <td
          class="css-0"
          title="--"
        >
          --
        </td>
        <td
          class="css-0"
          title="EcoRegistry"
        >
          EcoRegistry
        </td>
        <td
          class="css-0"
          title="AR-ACM0003: Afforestation and reforestation of lands except wetlands"
        >
          AR-ACM0003: Afforestation and reforestation of lands except wetlands
        </td>
        <td
          class="css-0"
          title="Agriculture; forestry and fishing"
        >
          Agriculture; forestry and fishing
        </td>
        <td
          class="css-0"
          title="Colombia"
        >
          Colombia
        </td>
        <td
          class="css-0"
          title="Not Started"
        >
          Not Started
        </td>
        <td
          class="css-rl3ocm"
        >
          <p
            class="chakra-text css-132cox5"
          >
            --
          </p>
        </td>
        <td
          class="css-rl3ocm"
          title="CDC_12_2_1_322_14_CO_1_1_2019_1-CDC_12_2_1_322_14_CO_1_1_2019_233"
        >
          CDC_12_2_1_322_14_CO_1_1_2019_1-CDC_12_2_1_322_14_CO_1_1_2019_233
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`renders correctly with hover 1`] = `
<div>
  <table
    class="chakra-table searchTable css-0"
  >
    <tbody
      class="css-0"
      data-testid="table-scroll"
    >
      <tr
        class="project-row-0 css-mii0oe hoverGreen"
        data-testid="unit-table-row"
      >
        <td
          class="css-0"
          title="Held"
        >
          Held
        </td>
        <td
          class="css-0"
          title="2020"
        >
          2020
        </td>
        <td
          class="css-0"
          title="129,037"
        >
          129,037
        </td>
        <td
          class="css-0"
          title="2023/04/14"
        >
          2023/04/14
        </td>
        <td
          class="css-0"
          title="EcoRegistry"
        >
          EcoRegistry
        </td>
        <td
          class="css-0"
          title="AR-ACM0003: Afforestation and reforestation of lands except wetlands"
        >
          AR-ACM0003: Afforestation and reforestation of lands except wetlands
        </td>
        <td
          class="css-0"
          title="Agriculture; forestry and fishing"
        >
          Agriculture; forestry and fishing
        </td>
        <td
          class="css-0"
          title="Colombia"
        >
          Colombia
        </td>
        <td
          class="css-0"
          title="Not Started"
        >
          Not Started
        </td>
        <td
          class="css-rl3ocm"
        >
          <p
            class="chakra-text css-132cox5"
          >
            --
          </p>
        </td>
        <td
          class="css-rl3ocm"
          title="CDC_29_7_1_321_14_CO_1_3_2020_BF_1-CDC_29_7_1_321_14_CO_1_3_2020_BF_4936"
        >
          CDC_29_7_1_321_14_CO_1_3_2020_BF_1-CDC_29_7_1_321_14_CO_1_3_2020_BF_4936
        </td>
      </tr>
      <tr
        class="project-row-1 css-mii0oe"
        data-testid="unit-table-row"
      >
        <td
          class="css-0"
          title="Retired"
        >
          Retired
        </td>
        <td
          class="css-0"
          title="2019"
        >
          2019
        </td>
        <td
          class="css-0"
          title="20,394"
        >
          20,394
        </td>
        <td
          class="css-0"
          title="--"
        >
          --
        </td>
        <td
          class="css-0"
          title="EcoRegistry"
        >
          EcoRegistry
        </td>
        <td
          class="css-0"
          title="AR-ACM0003: Afforestation and reforestation of lands except wetlands"
        >
          AR-ACM0003: Afforestation and reforestation of lands except wetlands
        </td>
        <td
          class="css-0"
          title="Agriculture; forestry and fishing"
        >
          Agriculture; forestry and fishing
        </td>
        <td
          class="css-0"
          title="Colombia"
        >
          Colombia
        </td>
        <td
          class="css-0"
          title="Not Started"
        >
          Not Started
        </td>
        <td
          class="css-rl3ocm"
        >
          <p
            class="chakra-text css-132cox5"
          >
            --
          </p>
        </td>
        <td
          class="css-rl3ocm"
          title="CDC_12_2_1_322_14_CO_1_1_2019_1-CDC_12_2_1_322_14_CO_1_1_2019_233"
        >
          CDC_12_2_1_322_14_CO_1_1_2019_1-CDC_12_2_1_322_14_CO_1_1_2019_233
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
